Failure converting Handshake generic covenant to Update

We see this somewhere after block 20,000

```
{"timestamp":"2025-12-02T18:20:55-05:00","level":"DEBUG","msg":"synced Handshake block","component":"main","hash":"000000000000003b9e651e9a155f262c3d8ac2af9fd4e3023fa6ac2882de1e16","prevHash":"000000000000001f5f81b46eabbe2e2c13e93f26d51a70c7f74e3b25526c8b67"}
panic: can't convert generic covenant to Update: EOF

goroutine 174 [running]:
github.com/blinklabs-io/cdnsd/handshake.(*GenericCovenant).Covenant(0xc0001387a0?)
        /home/agaffney/dev/blinklabs-io/cdnsd/handshake/covenant.go:112 +0x4e5
github.com/blinklabs-io/cdnsd/internal/indexer.(*Indexer).handshakeHandleSync(0x1484ce0?, 0xc00297fe60)
        /home/agaffney/dev/blinklabs-io/cdnsd/internal/indexer/handshake.go:137 +0x47f
```

--- a/handshake/covenant.go
+++ b/handshake/covenant.go
@@ -355,11 +355,13 @@ func NewRegisterCovenantFromGeneric(
 	// Decode height from bytes
 	ret.Height = binary.LittleEndian.Uint32(gc.Items[1])
 	// Decode resource data
-	tmpData, err := NewDomainResourceDataFromBytes(gc.Items[2])
-	if err != nil {
-		return nil, err
+	if len(gc.Items[2]) > 0 {
+		tmpData, err := NewDomainResourceDataFromBytes(gc.Items[2])
+		if err != nil {
+			return nil, err
+		}
+		ret.ResourceData = *tmpData
 	}
-	ret.ResourceData = *tmpData
 	return ret, nil
 }
 
@@ -391,11 +393,13 @@ func NewUpdateCovenantFromGeneric(
 	// Decode height from bytes
 	ret.Height = binary.LittleEndian.Uint32(gc.Items[1])
 	// Decode resource data
-	tmpData, err := NewDomainResourceDataFromBytes(gc.Items[2])
-	if err != nil {
-		return nil, err
+	if len(gc.Items[2]) > 0 {
+		tmpData, err := NewDomainResourceDataFromBytes(gc.Items[2])
+		if err != nil {
+			return nil, fmt.Errorf("decode domain resource data: %w", err)
+		}
+		ret.ResourceData = *tmpData
 	}
-	ret.ResourceData = *tmpData
 	return ret, nil
 }
 

--- a/handshake/domain.go
+++ b/handshake/domain.go
@@ -70,18 +70,19 @@ func (d *DomainResourceData) decode(data []byte) error {
 	var err error
 	// Version
 	if err = binary.Read(r, binary.LittleEndian, &d.Version); err != nil {
-		return err
+		return fmt.Errorf("read version: %w", err)
 	}
 	// Records
 	var recordType uint8
 	var record DomainRecord
+recordLoop:
 	for {
 		// Read record type
 		if err = binary.Read(r, binary.LittleEndian, &recordType); err != nil {
 			if errors.Is(err, io.EOF) {
 				break
 			}
-			return err
+			return fmt.Errorf("read record type: %w", err)
 		}
 		switch recordType {
 		case RecordTypeDS:
@@ -99,12 +100,14 @@ func (d *DomainResourceData) decode(data []byte) error {
 		case RecordTypeTEXT:
 			record = &TextDomainRecord{}
 		default:
-			return fmt.Errorf("unsupported record type %d", recordType)
+			// Stop processing on unknown record type
+			// This matches the behavior of hsd
+			break recordLoop
 		}
 		if record != nil {
 			err = record.decode(r)
 			if err != nil {
-				return err
+				return fmt.Errorf("decode record: %w", err)
 			}
 		}
 		d.Records = append(d.Records, record)

